src/v/raft/rpc_client_protocol.cc

```diff
@@ -89,4 +91,29 @@ ss::future<result<timeout_now_reply>> rpc_client_protocol::timeout_now(
       });
 }
 
+ss::future<bool> rpc_client_protocol::ensure_disconnect(model::node_id n) {
+    class Resetter {
+    public:
+        rpc::transport& transport;
+        Resetter(rpc::transport& t)
+          : transport(t) {}
+    };
+
+    return _connection_cache.local()
+      .with_node_client<Resetter>(
+        _self,
+        ss::this_shard_id(),
+        n,
+        std::chrono::milliseconds(100),
+        [](Resetter r) {
+            if (r.transport.is_valid()) {
```

Comment:
you can always just shutdown() no need for bool-check

Reply:
Returning the bool is a log hygiene thing, so that heartbeat_manager can log when it has really torn down a connection, rather than every time it calls `ensure_disconnect` -- I've tweaked this function to make it more obvious that we're really only capturing is_valid as an advisory thing for the caller.